lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -291,9 +291,15 @@ collectLoads(Operand *addressUse, CopyAddrInst *originalCopy,
 /// generates). Instead we guarantee that all normal uses are within the block
 /// of the temporary and look for the last use, which effectively ends the
 /// lifetime.
-bool TempRValueOptPass::checkNoSourceModification(
+///
+/// Return the instrution which is the last use (or copyInst if there are no
+/// uses).
+/// Returns a nullptr, if there is a source modification before the last use.
+SILInstruction *TempRValueOptPass::checkNoSourceModification(
```

Comment:
Can we rename this now/make the comment clearer. My thought is the comment should be something like:

```
Given the list of uses of the target copy_addr that was found by
walking def-use chains, walk the block where those uses were
defined to prove that the source is not written to before the
last use of that temporary.

Returns the last load in the use block that we found or the
underlying copy_addr we are trying to eliminate if we do not have
any uses. Returns nullptr, if there are any source modifications
before we find the last use in our copy_addr use list.

NOTE: Unfortunately, we cannot simply use the destroy points of
the temporary as the lifetime end, because these destroys may be
in a different basic block due to SILGen cleaning up temporaries
at end of lexical scopes rather than at last use. To work around
this, our caller guarantees that all normal uses are within the
block of the temporary so that we can just look for the last
non-destroying use, which we can use to model the end of the
lifetime of the temporary.
```

I would also change the name of the function to be mean something that implies something more along the lines of the comment. My thought is that the function name should have something that suggests we are performing a walk of the temporaries uses, not just checking no source modification.

Reply:
I renamed it and change the comment (though I kept it smaller than you suggested)